File: UI/pc-part-picker/src/model/Processor.ts
import { PcPart } from './Pcpart';
import { ManufacturerType } from "./ManufacturerType";
import { SeriesType } from "./SeriesType";
import { SocketType } from "./SocketType";

export interface Processor extends PcPart {
    socketTypeId: number | null;
    coreCount: number | null;
    coreClock: number | null;
    boostClock: number | null;
    tdp: number | null;
    graphics: string | null;
    smt: boolean | null;
    manufacturerTypeId: number | null;
    seriesTypeId: number | null;
    manufacturerType: string | null;
    seriesType: string | null;
    socketType: string | null;
}

File: API/PCPartPicker/Models/Processor.cs
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Processor
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public int? SocketTypeId { get; set; }

    public int? CoreCount { get; set; }

    public decimal? CoreClock { get; set; }

    public decimal? BoostClock { get; set; }

    public int? Tdp { get; set; }

    public string? Graphics { get; set; }

    public bool? Smt { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public int? SeriesTypeId { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual ManufacturerType? ManufacturerType { get; set; }

    public virtual SeriesType? SeriesType { get; set; }

    public virtual SocketType? SocketType { get; set; }
}
